core/src/main/java/org/apache/calcite/sql/type/SqlTypeUtil.java

```diff
@@ -294,6 +294,33 @@ public static boolean containsNullable(RelDataType type) {
     return false;
   }
 
+  /**
+   * Recreates a RelDataType with the type keep same as the param sourceRelDataType and
+   * nullability keep same as the param targetRelDataType.
```

Comment:
What about rephrasing the comment as follows:
```suggestion
   * Creates a RelDataType having the same type of the sourceRelDataType,
   * and the same nullability as the targetRelDataType.
```

Reply:
Looks good, I will fix it.